retriever/lib/engine.py

```diff
@@ -60,11 +61,12 @@ def disconnect(self):
             self.connection.close()
             self._connection = None
             self._cursor = None
-
+    
+    
     def get_connection(self):
         """This method should be overloaded by specific implementations
         of Engine."""
-        pass
+        raise NotImplementedError
```

Comment:
Based on the comment this should be overridden not overloaded, 
"""This method should be overriddend by specific implementations
        of Engine."""